### PR TITLE
2069: Add group and collaborator methods to Forge and HostedRepository

### PR DESCRIPTION
**Pull request by erikj79**
This patch adds some new methods on Forge and HostedRepository to interact with forge groups and repository collaborators respectively. It also adds a lot more possible customization to the test class `CensusBuilder` as well as adding some missing `final` declarations in test classes. All the new API calls have manual tests added. It's quite large for what it is, but isn't really affecting any current production code.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-2069](https://bugs.openjdk.org/browse/SKARA-2069): Add group and collaborator methods to Forge and HostedRepository (**Enhancement** - P4)

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [17f3f8b4](https://git.openjdk.org/skara/pull/1573/files/17f3f8b43b10148b11df6e10d2ef3d4e62caf518)
 * [Zhao Song](https://openjdk.org/census#zsong) (@zhaosongzs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1573/head:pull/1573` \
`$ git checkout pull/1573`

Update a local copy of the PR: \
`$ git checkout pull/1573` \
`$ git pull https://git.openjdk.org/skara.git pull/1573/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1573`

View PR using the GUI difftool: \
`$ git pr show -t 1573`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1573.diff">https://git.openjdk.org/skara/pull/1573.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1573#issuecomment-1771725900)

### CODE REVIEW

**Comment by zhaosongzs** [comment]
I took a quick look, and it looks good, but we need to update the copyright and add copyright for new files.

File: forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabRestApiTest.java [hunk 1]
```diff
@@ -66,13 +75,7 @@ void testReviews() throws IOException {
     }
 
     @Test
-    void testFilesUrl() throws IOException {
```
Comment (magicus) [inline forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabRestApiTest.java:69]: Nice cleanup...

**Review by magicus — APPROVED** [review]
Looks good to me.

**Review by zhaosongzs — APPROVED** [review]
Looks very good!

**Pull request closed without merging**